Upstream status: unappropriate

Signed-off-by: Maksim Sisov <msisov@igalia.com>
---
From 212bcff404ef9951b9b5cfb49f155645e24ab503 Mon Sep 17 00:00:00 2001
From: Maksim Sisov <msisov@igalia.com>
Date: Wed, 12 Sep 2018 12:42:34 +0300
Subject: [PATCH] Rotate gcc toolchain's build flags

The patch is taken from https://github.com/Igalia/meta-browser/pull/9.

By this, we can ensure chromium libs are compiled with right flags
and those are not overriden with AGL ones.
---
 build/toolchain/gcc_toolchain.gni | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/build/toolchain/gcc_toolchain.gni b/build/toolchain/gcc_toolchain.gni
index adadb7447e1d..b2a81154c65b 100644
--- a/build/toolchain/gcc_toolchain.gni
+++ b/build/toolchain/gcc_toolchain.gni
@@ -281,7 +281,7 @@ template("gcc_toolchain") {
 
     tool("cc") {
       depfile = "{{output}}.d"
-      command = "$cc -MMD -MF $depfile ${rebuild_string}{{defines}} {{include_dirs}} {{cflags}} {{cflags_c}}${extra_cppflags}${extra_cflags} -c {{source}} -o {{output}}"
+      command = "$cc -MMD -MF $depfile ${extra_cppflags}${extra_cflags} ${rebuild_string}{{defines}} {{include_dirs}} {{cflags}} {{cflags_c}} -c {{source}} -o {{output}}"
       depsformat = "gcc"
       description = "CC {{output}}"
       outputs = [
@@ -291,7 +291,7 @@ template("gcc_toolchain") {
 
     tool("cxx") {
       depfile = "{{output}}.d"
-      command = "$cxx -MMD -MF $depfile ${rebuild_string}{{defines}} {{include_dirs}} {{cflags}} {{cflags_cc}}${extra_cppflags}${extra_cxxflags} -c {{source}} -o {{output}}"
+      command = "$cxx -MMD -MF $depfile ${extra_cppflags}${extra_cxxflags} ${rebuild_string}{{defines}} {{include_dirs}} {{cflags}} {{cflags_cc}} -c {{source}} -o {{output}}"
       depsformat = "gcc"
       description = "CXX {{output}}"
       outputs = [
-- 
2.11.0

